Use block distance in portal search radius

phemmer opened the pull request:
Previously when searching for a portal, the server would use the search radius to determine which chunks it needs to search, and then would consider any portal within those chunks as a match, even if it is outside the search radius.
This changes it so that the portal list is filtered to only include portals within the configured search radius (ignoring the Y coordinate).

fixes #3583

MrIvanPlays commented on Spigot-Server-Patches/0551-Use-block-distance-in-portal-search-radius.patch:
```diff
@@ -0,0 +1,47 @@
+-import java.util.List;
+-import java.util.Optional;
+-import java.util.Random;
++import java.util.*;
```
Minimal diff. Also * import

phemmer replied:
fixed

aikar commented on Spigot-Server-Patches/0551-Use-block-distance-in-portal-search-radius.patch:
```diff
@@ -0,0 +1,43 @@
+@@ -1,5 +1,6 @@
+ package net.minecraft.server;
+ 
++import java.util.Collections;
```
inline import please

aikar commented on Spigot-Server-Patches/0551-Use-block-distance-in-portal-search-radius.patch:
```diff
@@ -0,0 +1,43 @@
+-        }));
++        // Paper start - filter by searchRadius
++        list.removeIf(villagePlaceRecord -> {
++            BlockPosition portalPosition = villagePlaceRecord.f();
```
This is a behavioral change vs Vanilla unless I'm missing something? So we need a config option for this.

However, I'd put it with the existing search radius configs block.

If your ok with not having authorship on the patch file itself, could merge with the previous.
But otherwise ok with a separate patch, just bundle the config next to the others.

phemmer replied:
Done.

phemmer committed:
Use block distance in portal search radius

Previously when searching for a portal, the server would use the search radius to determine which chunks it needs to search, and then would consider any portal within those chunks as a match, even if it is outside the search radius.
This changes it so that the portal list is filtered to only include portals within the configured search radius (ignoring the Y coordinate).

Proximyst commented on Spigot-Server-Patches/0051-Add-configurable-portal-search-radius.patch:
```diff
@@ -69,3 +71,24 @@ index 8eacfc66dc04bba89c8e1c03746c67068e07718b..f6ec165f9c7d83698100c85b1d4bf6b4
+-            return villageplacerecord.f().getY();
+-        }));
++        // Paper start - filter by searchRadius
++        if (world.paperConfig.portalSearchRadiusExact)
```
This would be much more readable with braces or whitespace following the statement

Proximyst commented on Spigot-Server-Patches/0051-Add-configurable-portal-search-radius.patch:
```diff
@@ -69,3 +71,24 @@ index 8eacfc66dc04bba89c8e1c03746c67068e07718b..f6ec165f9c7d83698100c85b1d4bf6b4
++        // Paper start - filter by searchRadius
++        if (world.paperConfig.portalSearchRadiusExact)
++            list.removeIf(villagePlaceRecord -> {
++                BlockPosition portalPosition = villagePlaceRecord.f();
```
OBFHELPER

Proximyst commented on Spigot-Server-Patches/0051-Add-configurable-portal-search-radius.patch:
```diff
@@ -69,3 +71,24 @@ index 8eacfc66dc04bba89c8e1c03746c67068e07718b..f6ec165f9c7d83698100c85b1d4bf6b4
++            });
++        Optional<VillagePlaceRecord> optional = list.size() == 0 ? Optional.empty() : Optional.of(
++            java.util.Collections.min(list, Comparator.<VillagePlaceRecord>comparingDouble((villageplacerecord) -> {
++                return villageplacerecord.f().j(blockposition);
```
It'd definitely be possible to not touch these lines as they're unmodified.
